acra-translator/go-translator-demo/handlers/user.go

```diff
@@ -0,0 +1,431 @@
+package handlers
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+
+	"cossacklabs/acra-translator-demo/acra-translator"
+	"cossacklabs/acra-translator-demo/models"
+	"cossacklabs/acra-translator-demo/pkg/errors"
+	"cossacklabs/acra-translator-demo/pkg/http/render"
+	"cossacklabs/acra-translator-demo/repositories"
+)
+
+type User struct {
+	userRepo         repositories.User
+	translatorClient acra_translator.Client
+}
+
+func NewUser(userRepo repositories.User, client acra_translator.Client) User {
+	return User{
+		userRepo,
+		client,
+	}
+}
+
+// Save
+//
+//	@Summary
+//	@Accept		json
+//	@Produce	json
+//
+//	@Router		/users [POST]
+//
+//	@Param		user_payload	body		models.User	true	"user payload"
+//
+//	@Success	200				{object}	render.SuccessResponse{data=models.User}
+//	@Failure	400,404,500		{object}	render.ErrorResponse{error=render.Result}
+//
+//	@tags		User Management
+func (u User) Save(w http.ResponseWriter, r *http.Request) {
+	var user models.User
+	err := u.parseFromRequest(r, &user)
+	if err != nil {
+		render.Error(w, r, errors.NewInvalidInputError(fmt.Sprintf("unable to parse api user model - %s", err.Error())))
+		return
+	}
+
+	if err := user.Validate(); err != nil {
+		render.Error(w, r, errors.NewInvalidInputError(fmt.Sprintf("invalid input provided - %s", err.Error())))
+		return
+	}
+
+	// Here is the example if synchronous call to AcraTranslator.
+	// It can be simplified via using the BulkAPI supported by AcraTranslator Enterprise version
```

Comment:
let's leave url for further reading - https://docs.cossacklabs.com/acra/guides/integrating-acra-translator-into-new-infrastructure/http_api/#bulk-processing-api-enterprise